mlx_lm/quant/utils.py

```diff
@@ -47,13 +46,22 @@ def replace_linear_with_quant_linear(model, quant_method = "bitnet", modules_to_
             new_layer = QUANT_LINEAR_MAPPING[quant_method](in_features, out_features, bias=bias, invert_weight_scales=True)
 
             # Replace the layer in the model
-            quantize_layers.append((name, new_layer))
+            if fused_qkv and not any(name.endswith(suffix) for suffix in ["q_proj", "k_proj", "v_proj", "o_proj"]):
+                quantize_layers.append((name, new_layer))
+            elif not fused_qkv:
+                quantize_layers.append((name, new_layer))
+        if fused_qkv and name not in modules_to_not_convert and module.__class__.__name__ == "Attention":
+            # Replace Attention layers with BitLinearFusedAttention
+            from mlx_lm.models.bitlinear_layers import BitFusedAttention
+
+            new_module = BitFusedAttention(model.args, invert_weight_scales=True)
+            quantize_layers.append((name, new_module))
     if len(quantize_layers) > 0:
         model.update_modules(tree_unflatten(quantize_layers))
     return model
 
 
-def apply_hf_quantization(model, config):
+def apply_hf_quantization(model, config, weights):
```

Comment:
Please ensure the tests bitnet_llama tests are passing. Because I use this function.